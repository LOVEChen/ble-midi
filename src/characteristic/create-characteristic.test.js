/* eslint-env jest */
jest.mock('bleno/lib/characteristic');

test('should createCharacteristic()', () => {
  const mockCharacteristic = require('bleno/lib/characteristic');
  const createCharacteristic = require('./create-characteristic').default;
  const characteristic = createCharacteristic();
  expect(mockCharacteristic.mock.calls).toMatchSnapshot();
  expect(characteristic).toBeInstanceOf(mockCharacteristic);
});
