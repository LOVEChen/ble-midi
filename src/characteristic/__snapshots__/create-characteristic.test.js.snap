// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should createCharacteristic() 1`] = `
Array [
  Array [
    Object {
      "descriptors": Array [],
      "onIndicate": [Function],
      "onNotify": [Function],
      "onReadRequest": [Function],
      "onSubscribe": [Function],
      "onUnsubscribe": [Function],
      "onWriteRequest": [Function],
      "properties": Array [
        "read",
        "write",
        "writeWithoutResponse",
        "notify",
      ],
      "secure": Array [
        "read",
        "write",
        "writeWithoutResponse",
        "notify",
      ],
      "uuid": "7772E5DB-3868-4112-A1A9-F2669D106BF3",
      "value": null,
    },
  ],
]
`;
